cli/azd/pkg/project/ai_helper.go

```diff
@@ -0,0 +1,713 @@
+	modelVersionsClient   *armmachinelearning.ModelVersionsClient
+	endpointsClient       *armmachinelearning.OnlineEndpointsClient
+	deploymentsClient     *armmachinelearning.OnlineDeploymentsClient
+	initialized           bool
+}
+
+// NewAiHelper creates a new instance of AiHelper
+func NewAiHelper(
+	env *environment.Environment,
+	clock clock.Clock,
+	credentialProvider account.SubscriptionCredentialProvider,
+	pythonBridge ai.PythonBridge,
+	workspacesClient *armmachinelearning.WorkspacesClient,
+	envContainersClient *armmachinelearning.EnvironmentContainersClient,
+	envVersionsClient *armmachinelearning.EnvironmentVersionsClient,
+	modelContainersClient *armmachinelearning.ModelContainersClient,
+	modelVersionsClient *armmachinelearning.ModelVersionsClient,
+	endpointsClient *armmachinelearning.OnlineEndpointsClient,
+	deploymentsClient *armmachinelearning.OnlineDeploymentsClient,
+) AiHelper {
+	return &aiHelper{
+		env:                   env,
+		clock:                 clock,
+		pythonBridge:          pythonBridge,
+		workspacesClient:      workspacesClient,
+		envContainersClient:   envContainersClient,
+		envVersionsClient:     envVersionsClient,
+		modelContainersClient: modelContainersClient,
+		modelVersionsClient:   modelVersionsClient,
+		endpointsClient:       endpointsClient,
+		deploymentsClient:     deploymentsClient,
+	}
+}
+
+// RequiredExternalTools returns the required external tools for the AiHelper
+func (a *aiHelper) RequiredExternalTools(ctx context.Context) []tools.ExternalTool {
+	return a.pythonBridge.RequiredExternalTools(ctx)
+}
+
+// Initialize initializes the AiHelper
+func (a *aiHelper) Initialize(ctx context.Context) error {
+	if a.initialized {
+		return nil
+	}
+
+	if err := a.pythonBridge.Initialize(ctx); err != nil {
+		return err
+	}
+
+	a.initialized = true
+	return nil
+}
+
+// EnsureWorkspace ensures that the workspace exists
+func (a *aiHelper) EnsureWorkspace(
+	ctx context.Context,
+	scope *ai.Scope,
+) error {
+	workspaceName := scope.Workspace()
+
+	workspaceResponse, err := a.workspacesClient.Get(
+		ctx,
+		scope.ResourceGroup(),
+		workspaceName,
+		nil,
+	)
+	if err != nil {
+		return err
+	}
+
+	if *workspaceResponse.Workspace.Name != workspaceName {
+		return err
+	}
+
+	return nil
+}
+
+// CreateEnvironmentVersion creates a new environment version
+func (a *aiHelper) CreateEnvironmentVersion(
+	ctx context.Context,
+	scope *ai.Scope,
+	serviceConfig *ServiceConfig,
+	config *ai.ComponentConfig,
+) (*armmachinelearning.EnvironmentVersion, error) {
+	yamlFilePath := filepath.Join(serviceConfig.Path(), config.Path)
+	_, err := os.Stat(yamlFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	environmentName, err := config.Name.Envsubst(a.env.Getenv)
+	if err != nil {
+		return nil, fmt.Errorf("failed parsing environment name value: %w", err)
+	}
+
+	if environmentName == "" {
+		environmentName = fmt.Sprintf("%s-environment", serviceConfig.Name)
```

Comment:
Do we hace env-names restrictions?  like, what if  `serviceConfig.Name` is super long?  or has invalid chars?

Reply:
I'll look into it.

Reply:
Docs don't define any naming rules for model or environment names but do for deployment names. 

**Model Naming**
https://learn.microsoft.com/en-us/azure/machine-learning/reference-yaml-model?view=azureml-api-2

**Environment Naming**
https://learn.microsoft.com/en-us/azure/machine-learning/reference-yaml-environment?view=azureml-api-2

**Deployment Naming**
https://learn.microsoft.com/en-us/azure/machine-learning/how-to-manage-quotas?view=azureml-api-2#azure-machine-learning-online-endpoints-and-batch-endpoints